drivers/i2c/i2c_nrfx_twim.c

```diff
@@ -137,20 +138,55 @@ static const struct i2c_driver_api i2c_nrfx_twim_driver_api = {
 	.transfer  = i2c_nrfx_twim_transfer,
 };
 
-static int init_twim(struct device *dev, const nrfx_twim_config_t *config)
+static int init_twim(struct device *dev)
 {
-
-	nrfx_err_t result = nrfx_twim_init(&get_dev_config(dev)->twim, config,
-					   event_handler, dev);
+	nrfx_err_t result = nrfx_twim_init(&get_dev_config(dev)->twim,
+					   &get_dev_config(dev)->config,
+					   event_handler,
+					   dev);
 	if (result != NRFX_SUCCESS) {
 		LOG_ERR("Failed to initialize device: %s",
-			    dev->config->name);
+			dev->config->name);
 		return -EBUSY;
 	}
 
+
 	return 0;
 }
 
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+static int twim_nrfx_pm_control(struct device *dev, u32_t ctrl_command,
```

Comment:
It looks that runtime configuration (i2c_api_configure_t) is not preserved when power mode is changed.